### PR TITLE
adds an api endpoint and graphile job to fix mismatched deposits

### PR DESCRIPTION
**Pull request by hughy**
## Summary

fixes mismatched deposits in a series of four queries:
1. soft-deletes events for deposits on forks or without a block in 'blocks'
2. upserts events for deposits on main chain but marked otherwise
3. sets deposits.main equal to blocks.main wherever mismatched
4. sets deposits.main to false for any deposits without a block in 'blocks'

a POST to '/deposits/fix_mismatches' enqueues a worker job to execute these queries.

## Testing Plan

added tests and tested locally

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: src/events/deposits.upsert.service.ts
Comment (NullSoldier) [inline src/events/deposits.upsert.service.ts:159]: nit: I think we can move this directly into the controller, like we do for refresh_points. I don't think we need this method.

File: src/events/deposits.upsert.service.ts
Comment (rohanjadvani) [inline src/events/deposits.upsert.service.ts:173]:
```suggestion
    return this.prisma.$queryRawUnsafe<
```

File: src/events/deposits.upsert.service.ts
Comment (rohanjadvani) [inline src/events/deposits.upsert.service.ts:211]: I think this is fine for now, but if we see any starvation or long running worker jobs maybe we can chunk this out down the line.

**Review by rohanjadvani — APPROVED** [review]
LGTM after lint is fixed

File: src/events/deposits.controller.ts
Comment (rohanjadvani) [inline src/events/deposits.controller.ts:76]: What do you think about `refresh`, for consistency?

Comment (hughy) [reply src/events/deposits.controller.ts:76]: I like it, I'll change the endpoint and rename the associated methods

File: src/events/deposits.upsert.service.ts
Comment (hughy) [reply src/events/deposits.upsert.service.ts:213]: I think it may be worth breaking this loop into separate jobs now -- there are over 80k mismatched deposits right now, so the first time we run this will almost surely be a long running job.

**Review by rohanjadvani — COMMENTED** [review]
This code is great, but I think we should add some unit tests before merging for the new service methods.

**Pull request merged**
